docs/dev/development.md

```diff
@@ -122,6 +122,57 @@ scons reduces the clutter from compiler setup.
 Additionally, the tool supports options to filter by directory and file names and specify a lower
 bound value to report.
 
+### Address Sanitizer
+
+To debug memory corruptions, leaks, and other issues, DAOS can be compiled with the
+[AddressSanitizer (ASan)](https://github.com/google/sanitizers/wiki/AddressSanitizer) library.
+
+#### Building with ASan
+
+To build instrumented libraries, executables, RPMs, etc., use the `SANITIZERS=address` flag with the
+`scons` command.  This option is also managed by the CI system through a Jenkins configuration
+variable.
+
+ASan is supported with both Clang and GCC compilers. However, some compiler-specific configurations,
```

Comment:
`However, ...`
Shall we move this info to the `compiler_setuo.py` script where actual values are proposed?

Reply:
> I think we should do both. I will add this info int the `compiler_setup.py` script.
> 
> * [x]  Add info about compiler custom setting and consequences in the `compiler_setup.py` script.

Fixed with commit f38fd44a0